src/main/java/org/sagebionetworks/web/client/SynapseJavascriptClient.java

```diff
@@ -901,6 +901,11 @@ public void getChallenges(String userId, Integer limit, Integer offset, AsyncCal
 		doGet(url, OBJECT_TYPE.ChallengePagedResults, callback);
 	}
 
+	public FluentFuture<Challenge> getChallengeForProject(String projectId) {
+		String url = getRepoServiceUrl() + ENTITY + "/" + projectId + "/" + CHALLENGE;
+		return getFuture(cb -> doGet(url, OBJECT_TYPE.Challenge, cb));
+	}
```

Comment:
I was going to try put this method in the ChallengeClientAsync, but I think that implementation is generated.

Reply:
ChallengeClientAsync is part of one of the older GWT RPC servlet endpoints (see http://www.gwtproject.org/doc/latest/DevGuideServerCommunication.html for more info).  Perhaps surprisingly, this was not generated.
I prefer to now go directly to the backend via this SynapseJavascriptClient.